common/src/main/java/com/microsoft/identity/common/internal/platform/AndroidDeviceKeyManager.java

```diff
@@ -22,170 +22,57 @@
-    private final KeyStore mKeyStore;
-    @Getter
-    private final String mKeyAlias;
-    private final Supplier<byte[]> mThumbprintSupplier;
+    private static final String TAG = AndroidDeviceKeyManager.class.getSimpleName();
 
     @Builder
-    public DeviceKeyManager(@NonNull final KeyStore keyStore, @NonNull final String keyAlias,
-                            @NonNull final Supplier<byte[]> thumbprintSupplier) throws KeyStoreException {
-        this.mKeyAlias = keyAlias;
-        this.mThumbprintSupplier = thumbprintSupplier;
-        this.mKeyStore = keyStore;
-    }
-
-
-    @Override
-    public boolean exists() {
-        boolean exists = false;
-
-        try {
-            exists = mKeyStore.containsAlias(mKeyAlias);
-        } catch (final KeyStoreException e) {
-            Logger.error(
-                    TAG,
-                    "Error while querying KeyStore",
-                    e
-            );
-        }
-
-        return exists;
-
-    }
-
-    @Override
-    public boolean hasThumbprint(@NonNull final byte[] thumbprint) {
-        return Arrays.equals(thumbprint, mThumbprintSupplier.get());
-    }
-
-    @Override
-    public Date getCreationDate() throws ClientException {
-        try {
-            return mKeyStore.getCreationDate(mKeyAlias);
-        } catch (final KeyStoreException e) {
-            Logger.error(
-                    TAG,
-                    "Error while getting creation date for alias " + mKeyAlias,
-                    e
-            );
-            throw new ClientException(ClientException.KEYSTORE_NOT_INITIALIZED, e.getMessage(), e);
-        }
-    }
-
-    @Override
-    public boolean clear() {
-        boolean deleted = false;
-
-        try {
-            mKeyStore.deleteEntry(mKeyAlias);
-            deleted = true;
-        } catch (final KeyStoreException e) {
-            Logger.error(
-                    TAG,
-                    "Error while clearing KeyStore",
-                    e
-            );
-        }
-
-        return deleted;
-    }
-
-
-    /**
-     * Retrieve the entry stored in this particular alias.
-     *
-     * @return the Entry in question, or null if the entry does not exist.
-     * @throws UnrecoverableEntryException
-     * @throws NoSuchAlgorithmException
-     * @throws KeyStoreException
-     */
-    @SuppressWarnings("unchecked")
-    @Override
-    public K getEntry() throws UnrecoverableEntryException, NoSuchAlgorithmException, KeyStoreException {
-        return (K) mKeyStore.getEntry(mKeyAlias, null);
+    public AndroidDeviceKeyManager(@NonNull final KeyStore keyStore, @NonNull final String keyAlias) throws KeyStoreException {
+        super(keyStore, keyAlias, null);
     }
 
     @Override
-    public void importKey(@NonNull final byte[] jwk, @NonNull final String algorithm) throws ClientException {
+    public void storeAsymmetricKey(PrivateKey privateKey, Certificate[] certChain) {
```

Comment:
What is the difference between Android vs Linux? why do we need this in one, but not the other?

Reply:
(Does this also mean that it doesn't belong in the abstract class)?

Reply:
Android Keystore stores the key in the keystore during key generation and we don't need to explicitly invoke `setKeyEntry`. Linux keystore, and more specifically BKS keystore doesn't do that and we need to explicitly invoke `setKeyEntry` and that's where this method comes in. I implemented the functionality in the abstract class because if we add more implementations in the future then they probably work similar to BKS implementation because the way I see it is that Android gets special treatment. (Most keystores require calling setKeyEntry to save the key into the key store)

Reply:
Could you please add this comment here as a future reference? Thanks!

Reply:
Addressed